Sources/TokamakCore/Views/Containers/ForEach.swift

```diff
@@ -80,8 +80,49 @@ public extension ForEach where Data == Range<Int>, ID == Int {
 
 extension ForEach: ParentView {
   public var children: [AnyView] {
-    data.map { AnyView(content($0)) }
+    data.map { AnyView(IDView(content($0), id: $0[keyPath: id])) }
   }
 }
 
 extension ForEach: GroupView {}
+
+struct _IDKey: EnvironmentKey {
+  static let defaultValue: AnyHashable? = nil
+}
+
+extension EnvironmentValues {
+  public var _id: AnyHashable? {
+    get {
+      self[_IDKey.self]
+    }
+    set {
+      self[_IDKey.self] = newValue
+    }
+  }
+}
+
+public protocol _AnyIDView {
+  var anyId: AnyHashable { get }
+}
```

Comment:
This protocol and the property aren't used anywhere, I also don't see it in the upstream API?

Reply:
Ah, sorry. I implemented `IDView` when I was trying different things to store the selection state of the sidebar.

I made `_AnyIDView` so you could check if the View has an ID, and access it. I think it'd be necessary for ForEach/List when animations are added so transitions can be made on the items. I can move ID view to a separate PR if that'd be better?

Reply:
No worries, keeping it this PR is fine, IMO.